File: detectors/fixtures/go/negative/reverted_guard_after_revert.go
// Fixture: state AFTER the revert commit (the guard fn was removed).
// validateTransferLeavesStatus is gone — this is the audit-pin state for
// the Go reverted-guard synthetic test. Live code is unprotected.

package transfer

// Vault holds a simple balance.
type Vault struct {
	Balance uint64
}

// Transfer executes a transfer without validating leaf status.
// Guard was here; reverted by "Trust mitigations" commit.
func (v *Vault) Transfer(amount uint64, leafStatus string) error {
	v.Balance += amount
	return nil
}

File: detectors/fixtures/rust/negative/reverted_guard_after_revert.rs
// Fixture: state AFTER the revert commit (the guard fn was removed).
// validate_supply_cap is gone — this is the audit-pin state for the Rust
// reverted-guard synthetic test. The live code has no supply-cap guard.

pub struct Vault {
    pub balance: u64,
}

impl Vault {
    pub fn deposit(&mut self, amount: u64, _cap: u64) -> Result<(), &'static str> {
        // Guard was here; reverted by "Trust mitigations" commit.
        self.balance += amount;
        Ok(())
    }
}
